ubcpi/persistence.py

```diff
@@ -0,0 +1,148 @@
+from submissions import api as sub_api
+
+"""
+answer_item has to be in the format:
+{
+    'answers': [
+        {
+            'vote': 'some option',
+            'rationale': 'some rationale'
+        },
+        {
+            'vote': 'some option',
+            'rationale': 'some rationale'
+        }
+    ]
+}
+"""
+
+ANSWER_LIST_KEY = 'answers'
+
+VOTE_KEY = 'vote'
+RATIONALE_KEY = 'rationale'
+
+
+def get_answers_for_student(student_item):
+    """
+    Retrieve answers from backend for a student and question
+
+    Args:
+        student_item (dict): The location of the problem this submission is
+            associated with, as defined by a course, student, and item.
+
+    Returns:
+        Answers: answers for the student
+    """
+    submissions = sub_api.get_submissions(student_item)
+    if not submissions:
+        return Answers()
+
+    latest_submission = submissions[0]
+    latest_answer_item = latest_submission.get('answer', {})
+    return Answers(latest_answer_item.get(ANSWER_LIST_KEY, []))
+
+
+def add_answer_for_student(student_item, vote, rationale):
+    """
+    Add an answer for a student to the backend
+
+    Args:
+        student_item (dict): The location of the problem this submission is
+            associated with, as defined by a course, student, and item.
+        vote (int): the option that student voted for
+        rationale (str): the reason why the student vote for the option
+    """
+    answers = get_answers_for_student(student_item)
+    answers.add_answer(vote, rationale)
+
+    sub_api.create_submission(student_item, {
+        ANSWER_LIST_KEY: answers.get_answers_as_list()
+    })
+
+
+class Answers:
+    """
+    The class that encapsulate the student answers
```

Comment:
Sorry for the confusion. Each student are required to submit answer twice. One for original, one for revised. This class encapsulate both, but just for one student. I updated the docstring for the class, hopefully it will clear things up a little bit.